### DISCUSSION TITLE
blog/change-desktop-wallpaper-with-powershell/

### DISCUSSION THREAD

**Discussion by giscus** [hanpq/getps.dev#18] [Comments]
# blog/change-desktop-wallpaper-with-powershell/

I’ve seen numerous forums and blog articles trying to to change desktop wallpaper in windows, none of which works reliably.

https://getps.dev/blog/change-desktop-wallpaper-with-powershell/

**Comment by ltalbert81** [comment]
Hello there! I've been trying to set the wallpaper on a remote pc.

I start a pssession and try this script:

$MyWallpaper="C:\users\smcpuser\pictures\fondos\2024\vishing.jpeg"
$code = @' 
using System.Runtime.InteropServices; 
namespace Win32{ 
    
     public class Wallpaper{ 
        [DllImport("user32.dll", CharSet=CharSet.Auto)] 
         static extern int SystemParametersInfo (int uAction , int uParam , string lpvParam , int fuWinIni) ; 
         
         public static void SetWallpaper(string thePath){ 
            SystemParametersInfo(20,0,thePath,3); 
         }
    }
 } 
'@

add-type $code 
[Win32.Wallpaper]::SetWallpaper($MyWallpaper)

It works Locally, but not remotely when I start a pssession.

Could you help me on this?

Sorry for my english.
